### PR TITLE
#234 Improve action handling flow

### PR DESCRIPTION
**Pull request by tortmayr**
Introduce `GLSPVscodeExtensionActionHandler` that provides the delegation/dispatching of local actions from the webview to the vsode extension. This enables the implementation of action handlers that require access  to the vscode API and/or node backend.

Also:
+ cleanup and centralize command registration into the `GLSPEditorContext` instead of the editor provider.
+ Add `registerActionHandlers()` to `GLSPEditorContext` method that enables the registration of additional extension action handlers in the user space
- Move `extensionPrefix` from constructor into abstract property
- Fix typos

### CODE REVIEW

File: vscode-integration-webview/src/extension-action-handler.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * of the webview. This enables the implementation of action handlers that require access
+ * to the vscode API and/or node backend.
+ */
+
+export class GLSPVscodeExtensionActionHandler implements IActionHandler, IActionHandlerInitializer {
+
+    constructor(protected readonly actionKinds: string[],
+        protected readonly diagramIdentifier: SprottyDiagramIdentifier) {
+    }
+    initialize(registry: ActionHandlerRegistry): void {
```
Comment (planger) [inline vscode-integration-webview/src/extension-action-handler.ts:31]:
```suggestion

    initialize(registry: ActionHandlerRegistry): void {
```

File: vscode-integration-webview/src/extension-action-handler.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import { Action, ActionHandlerRegistry, IActionHandler, IActionHandlerInitializer, ICommand } from 'sprotty';
+import { SprottyDiagramIdentifier } from 'sprotty-vscode-webview';
+import { vscodeApi } from 'sprotty-vscode-webview/lib/vscode-api';
+
+/**
+ * Delegates actions that should be handled inside of the glsp vscode extension instead
+ * of the webview. This enables the implementation of action handlers that require access
+ * to the vscode API and/or node backend.
+ */
+
```
Comment (planger) [inline vscode-integration-webview/src/extension-action-handler.ts:25]:
```suggestion
```

File: vscode-integration-webview/src/extension-action-handler.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    initialize(registry: ActionHandlerRegistry): void {
+        this.actionKinds.forEach(kind => registry.register(kind, this));
+    }
+
+    handle(action: Action): void | Action | ICommand {
+        if (this.actionKinds.includes(action.kind)) {
+            const message = {
+                clientId: this.diagramIdentifier.clientId,
+                action,
+                _localDispatch: true
```
Comment (planger) [inline vscode-integration-webview/src/extension-action-handler.ts:40]:
```suggestion
                __localDispatch: true
```

File: vscode-integration-webview/src/glsp-vscode-diagramserver.ts [hunk 4]
```diff
@@ -29,8 +29,8 @@ import { SelectionService } from '@eclipse-glsp/client/lib/features/select/selec
 import { inject } from 'inversify';
 import { isActionMessage, VscodeDiagramServer } from 'sprotty-vscode-webview';
 
-const receivedFromServerProperty = '__receivedFromServer';
-const localDispatchProperty = '_localDispatch';
+export const receivedFromServerProperty = '__receivedFromServer';
+export const localDispatchProperty = '_localDispatch';
```
Comment (planger) [inline vscode-integration-webview/src/glsp-vscode-diagramserver.ts:33]:
```suggestion
export const localDispatchProperty = '__localDispatch';
```

File: vscode-integration/src/glsp-diagram-editor-context.ts [hunk 5]
```diff
@@ -91,6 +101,27 @@ export abstract class GlspDiagramEditorContext extends Disposable {
         await this.onReady;
         return this._glspClient;
     }
+
+    protected registerCommands(): void {
+        this.registerActionCommand(GLSPCommand.FIT_TO_SCREEN, new FitToScreenAction([]));
+        this.registerActionCommand(GLSPCommand.CENTER, new CenterAction([]));
+        this.registerActionCommand(GLSPCommand.LAYOUT, new LayoutOperation());
+    }
```
Comment (planger) [inline vscode-integration/src/glsp-diagram-editor-context.ts:109]:
```suggestion
    }

```

**Review by planger — CHANGES_REQUESTED** [review]
Thanks Tobias, looks great to me! Just a few style suggestions inline.

@lforst Can you please test whether this works for your use cases?

File: vscode-integration-webview/src/extension-action-handler.ts (hunk 1 above)
Comment (tortmayr) [reply vscode-integration-webview/src/extension-action-handler.ts:31]: Done

File: vscode-integration/src/glsp-diagram-editor-context.ts (hunk 5 above)
Comment (tortmayr) [reply vscode-integration/src/glsp-diagram-editor-context.ts:109]: Done

**Comment by lforst** [comment]
> @lforst Can you please test whether this works for your use cases?

Works like a charm! Thanks for taking the time to implement this!

**Review by planger — APPROVED** [review]
Thanks, looks good to me! :+1:
Also, thanks @lforst for testing!

**Pull request merged**